model_inference_triage_pipeline/models/run_tf_models.py

```diff
@@ -0,0 +1,218 @@
+    )
+
+    _logger.info("Making predictions for probable security issues")
+    sec_pred_probs = sc_model.predict(
+        filtered_security_encoded_docs, batch_size=mc.BATCH_SIZE_PROB_SEC_BERT, verbose=0
+    )
+    sec_pred_probsr = sec_pred_probs.ravel()
+    sec_pred_labels = [1 if prob > 0.4 else 0 for prob in sec_pred_probsr]
+    _logger.info("Updating Security Model predictions in dataset")
+    df.loc[df.index.isin(sec_doc_idx), "security_model_flag"] = sec_pred_labels
+
+    _logger.info("Teardown security model")
+    del sc
+    del sc_model
+    gc.collect()
+
+    _logger.info("\n")
+
+    _logger.info("Keeping track of probable security issue rows")
+    subset_df = df[df["security_model_flag"] == 1]
+    prob_security_df_rowidx = np.array(subset_df.index)
+
+    sess = tf.Session()
+    BERT_MAX_SEQ_LEN = 512
+
+    _logger.info("Loading CVE Model")
+    bc = bcvec.BERTClassifier(
+        bert_model_path=cc.BASE_BERT_UNCASED_PATH, max_seq_length=BERT_MAX_SEQ_LEN
+    )
+    bc.build_model_architecture()
+
+    subset_df["norm_description"] = tn.pre_process_documents_parallel_bert(
+        documents=subset_df["description"].values
+    )
+    cve_encoded_docs = subset_df["norm_description"].values
+    _logger.info("Total CVE Docs Encoded: {n}".format(n=len(cve_encoded_docs)))
+    cve_doc_lengths = np.array([len(doc.split(" ")) for doc in cve_encoded_docs])
+    _logger.info("Removing bad docs with low tokens")
+    cve_doc_idx = np.argwhere(cve_doc_lengths >= 10).ravel()
+    filtered_cve_encoded_docs = cve_encoded_docs[cve_doc_idx]
+    _logger.info("Filtered CVE Docs Encoded: {n}".format(n=len(filtered_cve_encoded_docs)))
+
+    _logger.info("BERT text processing and feature engineering")
+    btp_obj = btp.BertTextProcessor(
+        tf_session=sess, bert_model_path=cc.BASE_BERT_UNCASED_PATH, max_seq_length=BERT_MAX_SEQ_LEN
+    )
+    btp_obj.create_bert_tokenizer()
+    btp_obj.convert_text_to_input_examples(filtered_cve_encoded_docs)
+    btp_obj.convert_examples_to_features()
+
+    _logger.info("Making predictions for probable CVE issues")
+    btp.initialize_vars(sess)
+    bc.load_model_weights(model_weights_path=cc.P2BERT_CVE_MODEL_WEIGHTS_PATH)
+
+    cve_pred_probs = bc.model_estimator.predict(
+        x=[btp_obj.input_ids, btp_obj.input_masks, btp_obj.segment_ids],
+        batch_size=mc.BATCH_SIZE_PROB_CVE_BERT,
+        verbose=1,
+    )
+    cve_pred_probsr = cve_pred_probs.ravel()
+    cve_pred_labels = [1 if prob > 0.5 else 0 for prob in cve_pred_probsr]
+    from collections import Counter
+
+    print(Counter(cve_pred_labels))
+    _logger.info("Updating CVE Model predictions in dataset")
+    prob_cve_idxs = prob_security_df_rowidx[cve_doc_idx]
+    df.loc[df.index.isin(prob_cve_idxs), "cve_model_flag"] = cve_pred_labels
+    _logger.info("Teardown CVE model")
+    _logger.info("\n")
+
+    del btp_obj
+    del bc
+    gc.collect()
+
+
+def run_gru_cve_model(df):
```

Comment:
Rather than adding code for gru and bert into single class , can we split it to two separate class and call run method  based on type?

Reply:
It's not a part of a single class, the reason they're grouped together is that both use Tensorflow as their runtime whereas the next model option would use Pytorch. So, in order to not import both together at the same time I did it this way.

Reply:
I have added a sample stub of how this will look post torch changes to show the proposed code structure, the actual changes I will push in a separate PR as it is new code. Do you think this runtime based modules design is fine or should I separate the modules too? The classes are already separate:

model_inference_triage_pipeline/models/bert_cve_classifier.py
model_inference_triage_pipeline/models/cve_dl_classifier.py
model_inference_triage_pipeline/models/security_dl_classifier.py

Reply:
Separate module makes more sense to me